### ISSUE TITLE
PHP8 support

### ISSUE THREAD

**Issue by affektde** [madebybob/php-number#7]
Are there any obstacles to make this package available for PHP8 as well?

I am testing it out briefly with a fork. Was going to start implementing `php-number` today, but had previously upgraded my project to PHP8.

**Comment by affektde** [comment]
Well this is what `composer test` says:

```phpunit
There was 1 error:

1) Number\Tests\NumberTest::testFormatNumber
TypeError: NumberFormatter::format(): Argument #1 ($num) must be of type int|float, string given

/mnt/c/Users/fsociety/Code/php-number/src/Formatter/Formatter.php:23
/mnt/c/Users/fsociety/Code/php-number/src/Number.php:21
/mnt/c/Users/fsociety/Code/php-number/tests/NumberTest.php:547

ERRORS!
Tests: 54, Assertions: 152, Errors: 1.
Script vendor/bin/phpunit handling the test event returned with error code 2
```

`$value` in `Formatter.php` seems to be a `string` here.
Would it be correct to cast `$value` here?

`->format((float) $value)`

**Comment by bobmulder** [comment]
Hi there @affektde,

thanks for pointing out. To be honest, I didn't do much testing with PHP 8 yet (has something to do with lack of time or so).

The specific error of the tests seems to be about the type of `NumberFormatter::DECIMAL` (which is an int according to [PHP](https://www.php.net/manual/en/class.numberformatter.php)). Maybe @jonmldr can say something useful about this.

I'll try to update all dependencies to support PHP 8 (and 8.1), and update Github Actions.

**Comment by jonmldr** [comment]
Hi @affektde! It's correct to cast `$value` to a float there.
We do cast the value to a float in the `formatMoney`, it should happen in `format` too.

Thank you for taking the effort to make php-number PHP 8 compatible! Are you able to submit a PR when you are finished?

**Comment by affektde** [comment]
#8 submitted my changes. Hope I did everything right.

**Comment by bobmulder** [comment]
Thank you @affektde. I've added PHP 8 to Github actions and 1.2.0 has been released with working 8.0 support.

Happy coding!
